lib/active_model/serializer/utils.rb

```diff
@@ -0,0 +1,42 @@
+module ActiveModel::Serializer::Utils
+  module_function
+
+    # converts the include hash to a standard format
+    # for constrained serialization recursion
+    #
+    # converts
+    #  [:author, :comments => [:author]] to
+    #  {:author => [], :comments => [:author]}
+    #
+    # and
+    #  [:author, :comments => {:author => :bio}, :posts => [:comments]] to
+    #  {:author => [], :comments => {:author => :bio}, :posts => [:comments]}
+    #
```

Comment:
The next line converts everything to an array, so my having the default an array, so if nothing is passed, the operation is a smidge faster.